src/userDashboard/dataFormat/BarGraphDataFormat.js

```diff
@@ -0,0 +1,221 @@
+
+  dates.forEach(day => {
+
+    var stringLegend = format(day, STRING_FORMAT); 
+
+    var year = getYear(day);
+    var month = getMonth(day);
+
+    var readingCount = readingData.has(year) ?
+                        readingData.get(year).has(month) ?
+                          readingData.get(year).get(month)
+                            : 0
+                            : 0 ;
+    
+    var exercisesCount = exercisesData.has(year) ?
+                          exercisesData.get(year).has(month) ?
+                            exercisesData.get(year).get(month)
+                              : 0
+                              : 0 ;
+                            
+    result.push({ date: stringLegend, reading_time: formatSeconds(readingCount, activeTimeFormatOption), exercises_time: formatSeconds(exercisesCount, activeTimeFormatOption)});
+
+  });
+
+  return result;
+
+}
+
+function getBarDataForYears(dataPerMonths, dateInYear, activeTimeFormatOption){
+  
+  var result = [];
+
+  const STRING_FORMAT = "yyyy"; 
+
+  var readingData = dataPerMonths.reading;
+  var exercisesData = dataPerMonths.exercises;
+
+  while(true){
+
+    var year = getYear(dateInYear);
+
+    var stringLegend = format(dateInYear, STRING_FORMAT); 
+
+    if ( !readingData.has(year) && !exercisesData.has(year) ){
+
+      break;
+
+    }
+
+    else{
+
+      var readingCount = 0;
+      var exercisesCount = 0;
+
+      if (readingData.has(year)){
+        for (const [ , value] of readingData.get(year).entries()) {
+          readingCount += value;
+        }
+      }
+
+      if (exercisesData.has(year)){
+        for (const [ , value] of exercisesData.get(year).entries()) {
+          exercisesCount += value;
+        }
+      }
+
+      result.push({ date: stringLegend, reading_time: formatSeconds(readingCount, activeTimeFormatOption), exercises_time: formatSeconds(exercisesCount, activeTimeFormatOption)});
+          
+      dateInYear = subYears(dateInYear, 1);
+
+    }
+  }
+
+  return result.reverse();
+
+}
+
+
+function calculateCountPerMonth_Activity(data){
+
+  var readingMap = calculatePerMonth(getMapData(data.reading));
+  var exercisesMap =  calculatePerMonth(getMapData(data.exercises));
+
+  return ({reading: readingMap, exercises: exercisesMap});
+
+}
+
+function getBarGraphData(data, countPerMonths, period, dateInPeriod, activeTimeFormatOption){
+    
+    switch(period) {
+      case PERIOD_OPTIONS.WEEK:
+        return getBarDataForWeek(data, dateInPeriod, activeTimeFormatOption); 
+      case PERIOD_OPTIONS.MONTH:
+        return getBarDataForMonth(data, dateInPeriod, activeTimeFormatOption); 
+      case PERIOD_OPTIONS.YEAR:
+        return getBarDataForYear(countPerMonths, dateInPeriod, activeTimeFormatOption); 
+      case PERIOD_OPTIONS.YEARS:
+        return getBarDataForYears(countPerMonths, dateInPeriod, activeTimeFormatOption); 
+      default:
+        return getBarDataForWeek(data, new Date(), activeTimeFormatOption); 

```

Comment:
why is this newDate() here?

Reply:
The function should never return the default ramification, I put it as a safety net. That's also why I created a new date object, so it would create and take the current date, if the date object passed was broken.

I can remove it, it does not make a difference.